docs/content/preview/releases/techadvisories/ta-24992.md

```diff
@@ -0,0 +1,41 @@
+---
+title: TA-24992
+headerTitle: Partial Streaming Stall After Tablet Split
+headcontent: 3 Dec 2024
+type: docs
+showRightNav: true
+cascade:
+  unversioned: true
+menu:
+  preview_releases:
+    identifier: ta-24992
+    weight: 1
+rightNav:
+  hideH2: true
+type: docs
+---
+
+
+|          Product           |  Affected Versions  |  Related Issues   | Fixed In |
+| :------------------------- | :------------------ | :---------------- | :------- |
+| [YugabyteDB gRPC (Debezium) Connector](https://docs.yugabyte.com/preview/explore/change-data-capture/using-yugabytedb-grpc-replication/debezium-connector-yugabytedb/)       | [dz.1.9.5.yb.grpc.2024.1](https://github.com/yugabyte/debezium-connector-yugabytedb/releases/tag/vdz.1.9.5.yb.grpc.2024.1) | {{<issue 24992 >}} | [dz.1.9.5.yb.grpc.2024.2](https://github.com/yugabyte/debezium-connector-yugabytedb/releases/tag/vdz.1.9.5.yb.grpc.2024.2) |
+
+## Description
+
+The YugabyteDB gRPC (Debezium) Connector may stop sending checkpoints to YugabyteDB, causing data streaming to stall for certain tablets.
+
+This issue is more likely to occur when a single task is configured to poll all the tablets of a table while the tablets are being split. If the checkpoint for a tablet is not updated and that tablet gets split, the connector fails to transition to the child tablets. As a result, no data from the child tablets will be streamed, causing the streaming process to halt for those tablets.
+
+## Mitigation
+
+To mitigate this issue, either upgrade the connector to [dz.1.9.5.yb.grpc.2024.2](https://github.com/yugabyte/debezium-connector-yugabytedb/releases/tag/vdz.1.9.5.yb.grpc.2024.2), **or** restart the connector using the following steps:
+
+
+1. Run the [get_unpolled_tablets.sh](https://gist.github.com/vaibhav-yb/8c7a7eca2599c250e8fd1080897b7e6f) script to get a list of unpolled tablets. For best results, run the script iteratively every 15 minutes and track the list of tablets it outputs.
+1. If there are tablets that are output across multiple runs of the script and never disappear, those tablets are stuck, and you should restart the connector.
+
+## Details
+
+When a Kafka commit occurs, the connector snapshots the current offsets across all source partitions. It identifies the highest offset for each tablet, logs these as checkpoints, and then updates YugabyteDB with the latest progress tracking information.
+
+The problem is how higher offsets are calculated: instead of merging new values into the existing global offset map, the map is completely overwritten. As a result, in cases of hot shards or actively written tablet ranges, some tablets might not receive a callback. In other words, offsets for certain tablets may be inadvertently skipped or filtered out.
```

Comment:
```suggestion
The problem lies in how higher offsets are calculated: instead of merging new values into the existing global offset map, the map is completely overwritten. As a result, in cases of hot shards or actively written tablet ranges, some tablets might not receive a callback. In other words, offsets for certain tablets may be inadvertently skipped or filtered out.
```